In [1]:
import pandas as pd
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

In [2]:
data=pd.read_csv('glass.csv')

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 214 entries, 0 to 213
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   RI      214 non-null    float64
 1   Na      214 non-null    float64
 2   Mg      214 non-null    float64
 3   Al      214 non-null    float64
 4   Si      214 non-null    float64
 5   K       214 non-null    float64
 6   Ca      214 non-null    float64
 7   Ba      214 non-null    float64
 8   Fe      214 non-null    float64
 9   Type    214 non-null    int64  
dtypes: float64(9), int64(1)
memory usage: 16.8 KB


In [4]:
x=data.values[:,0:9]
y=data.values[:,9]

In [5]:
x_train,x_test,y_train,y_test = train_test_split(x, y, test_size=0.3, random_state=42)
knn = KNeighborsClassifier(n_neighbors=2)
knn.fit(x_train,y_train)

KNeighborsClassifier(n_neighbors=2)

In [6]:
knn.score(x_train,y_train)

0.8053691275167785

In [7]:
knn.score(x_test,y_test) * 100

66.15384615384615

#### As we can see we have achieved 66.15% accuracy

### Classification

In [8]:
pred = knn.predict(x_test)

In [9]:
pred

array([1., 7., 1., 7., 2., 2., 1., 1., 2., 1., 2., 2., 1., 2., 6., 5., 6.,
       1., 1., 6., 2., 7., 7., 7., 1., 1., 1., 1., 5., 1., 2., 2., 3., 2.,
       1., 7., 5., 2., 1., 2., 2., 7., 3., 2., 1., 5., 6., 3., 6., 1., 1.,
       1., 1., 2., 1., 1., 2., 1., 5., 1., 2., 2., 1., 7., 2.])

In [10]:
print(classification_report(pred,y_test))

              precision    recall  f1-score   support

         1.0       0.84      0.64      0.73        25
         2.0       0.52      0.63      0.57        19
         3.0       0.25      0.33      0.29         3
         5.0       0.67      0.80      0.73         5
         6.0       0.67      0.40      0.50         5
         7.0       0.80      1.00      0.89         8

    accuracy                           0.66        65
   macro avg       0.62      0.63      0.62        65
weighted avg       0.69      0.66      0.66        65

